src/app/admin/packingSlotsTable/PackingSlotActions.ts

```diff
@@ -0,0 +1,162 @@
+import supabase from "@/supabaseClient";
+import { DatabaseError } from "@/app/errorClasses";
+import { GridRowId } from "@mui/x-data-grid";
+import { PackingSlotRow } from "@/app/admin/packingSlotsTable/PackingSlotsTable";
+
+export interface DBPackingSlotData {
+    primary_key?: string;
+    name: string;
+    is_shown: boolean;
+    order: number;
+}
+
+export interface DataGridPackingSlotRow {
+    id: string;
+    name: string;
+    is_shown: boolean;
+    order: number;
+}
+interface packingSlotTableRowData extends DataGridPackingSlotRow {
+    isNew: boolean;
+}
+
+export const fetchPackingSlots = async (): Promise<PackingSlotRow[]> => {
+    const { data, error } = await supabase.from("packing_slots").select().order("order");
+    if (error) {
+        throw new DatabaseError("fetch", "packing slots");
+    }
+
+    const processedData = data.map((row): PackingSlotRow => {
+        return {
+            name: row.name,
+            id: row.primary_key,
+            is_shown: row.is_shown,
+            order: row.order,
+            isNew: false,
+        };
+    });
+
+    return processedData;
+};
+
+const processRowData = (tableData: packingSlotTableRowData): DBPackingSlotData => {
```

Comment:
The function name could be a little more descriptive. How about `getDbPackingSlotFromTableRow(...)`?